pkg/cli/build.go

```diff
@@ -116,7 +112,11 @@ func addDockerfileFlag(cmd *cobra.Command) {
 }
 
 func addUseCogBaseImageFlag(cmd *cobra.Command) {
-	cmd.Flags().BoolVar(&buildUseCogBaseImage, "use-cog-base-image", true, "Use pre-built Cog base image for faster cold boots")
+	buildUseCogBaseImage = new(bool)
+	cmd.Flags().BoolVar(buildUseCogBaseImage, useCogBaseImageFlagKey, true, "Use pre-built Cog base image for faster cold boots")
+	if !cmd.Flags().Changed(useCogBaseImageFlagKey) {
+		buildUseCogBaseImage = nil
+	}
```

Comment:
Wait, does this work? My mental model of `cmd.Flags().Changed()` was that it evaluated in `RunE` when args were passed in, like a variable initialized and then set later.

Reply:
You are completely correct, let me find a different way to express this.

Reply:
I've got a fix and added some more tests, let me know if this is sufficient.